### DISCUSSION TITLE
Unable to remove NickServ from a channel once it's been invited

### DISCUSSION THREAD

**Discussion by ghost** [anope/anope#416] [Support] [answered]
**Description**

I decided to invite NickServ into one of the channels on my server (UnrealIRCd 6.1.6, anope 2.0.15), in order to use commands like !mute that initially worked fine...unfortunately, for reasons that I still don't understand, at the moment running a command like !help with NickServ in the channel results in no response.

As such I tried to remove NickServ from the channel and that's when I ran into trouble...a regular /kick didn't have the appropriate access to remove a service from the channel, even with server operator privileges...I tried /sapart as well and it reported success at having removed NickServ from the channel, despite the fact that it was still there. I am probably missing something, but it _seems_ as though once you invite NickServ or another one of the services to your channel, there is no practical way to remove it.

**Steps to reproduce the issue:**

1. On a server running UnrealIRCd and anope, invite, e.g. NickServ to one of the channels that you have the appropriate privileges for
2. Try to use any of the common methods to kick or otherwise remove NickServ from the channel

**Describe the results you received:**

I was unable to use any common method to remove NickServ from the channels, but I learned in working with anope's backups that the channels NickServ are a part of are "persisted" to anope.db. I restored a copy of this file from a backup and reloaded anope, at which point NickServ finally departed the channels I joined for good. I can't imagine that manually restoring the database from the backup is the _right_ way to do it, though.

**Describe the results you expected:**

As a server operator it should be possible to remove NickServ or the other services from a channel, in the event that a user invites them to a channel by mistake, it is a very difficult change to reverse.

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `services --version`:**

Anope-2.0.15 -- build #1, compiled 20:38:39 May 17 2024

**Comment by sadiepowell** [comment, marked as answer]
Inviting the bot assigns it to the channel. You can unassign it with `/msg BotServ UNASSIGN #channel`.

**Reply by ghost** [reply]
Thank you very much, we gave this a test on the server and I was successfully able to remove a bot from the channel with the command. Please excuse the report as a bug! I'm obviously not an experienced server admin and I still have a lot to learn 😄 

I have been able to successfully repeat inviting/removing bots from channels using this method without issue, so I would consider the "bug" as such to be closed completely! Thanks very much
